lib/finch/util.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Finch.Util do
+  @moduledoc false
+
+  def maybe_log_secrets(:https, socket) do
+    maybe_log_secrets(:https, System.get_env("SSLKEYLOGFILE"), socket)
+  end
+
+  def maybe_log_secrets(_scheme, _socket) do
+    :ok
+  end
+
+  defp maybe_log_secrets(:https, nil, _socket) do
+    :ok
+  end
+
+  defp maybe_log_secrets(:https, ssl_key_log_file, socket) do
+    # Note: not every ssl library version returns information for :keylog. By using `with` here,
+    # anything other than the expected return value is silently ignored.
+    with {:ok, [{:keylog, keylog_items}]} <- :ssl.connection_information(socket, [:keylog]),
+         {:ok, f} <- File.open(ssl_key_log_file, [:append]) do
+      try do
+        for keylog_item <- keylog_items do
+          :ok = IO.puts(f, keylog_item)
```

Comment:
How does this behave in a highly concurrent environment, ie: if there are many processes concurrently trying to append to the same file?

Reply:
Good point, I will fix that. Lines can be interleaved but obviously writing a line from `:keylog` has to be atomic.